src/pymedphys/dicom/_level2/header_tweaks.py

```diff
@@ -128,11 +128,16 @@ def adjust_rel_elec_density_cli(args):
     pydicom.write_file(args.output_file, new_ds)
 
 
-def RED_adjustment_map_from_structure_names(structure_names):
-
+def extract_RED_from_structure_name_regex():
```

Comment:
I didn't mean to suggest the regex could benefit from refactoring out as a *function*. I simply meant something like the following might help with understanding:

```python
def RED_adjustment_map_from_structure_names(structure_names):
    structure_name_regex = r'^.*RED\s*[=:]\s*(\d+\.?\d*)\s*$'
    structure_name_pattern = re.compile(structure_name_regex)
    adjustment_map = {
        structure: float(structure_name_pattern.match(structure).group(1))
        for structure in structure_names
        if structure_name_pattern.match(structure)
    }
    return adjustment_map

But I'm happy with either approach :)